src/openforms/contrib/customer_interactions/utils.py

```diff
@@ -0,0 +1,136 @@
+from collections.abc import Iterable, Mapping
+from itertools import groupby
+
+import structlog
+from openklant_client.types.methods.maak_klant_contact import MaakKlantContactResponse
+from openklant_client.types.resources.digitaal_adres import (
+    DigitaalAdres,
+    SoortDigitaalAdres,
+)
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.contrib.customer_interactions.client import (
+    get_customer_interactions_client,
+)
+from openforms.contrib.customer_interactions.models import (
+    CustomerInteractionsAPIGroupConfig,
+)
+from openforms.formio.typing.custom import DigitalAddress
+from openforms.forms.models import FormVariable
+from openforms.prefill.contrib.customer_interactions.typing import (
+    CommunicationChannel,
+    SupportedChannels,
+)
+from openforms.submissions.models import Submission
+from openforms.typing import JSONValue
+from openforms.variables.constants import FormVariableSources
+
+logger = structlog.stdlib.get_logger(__name__)
+
+ADDRESS_TYPES_TO_CHANNELS: Mapping[SoortDigitaalAdres, SupportedChannels] = {
+    "email": "email",
+    "telefoonnummer": "phoneNumber",
+}
+
+
+def transform_digital_addresses(
+    digital_addresses: Iterable[DigitaalAdres],
+    configured_address_types: list[SupportedChannels],
+) -> list[CommunicationChannel]:
+    """
+    Filter and group digital addresses.
+
+    This function:
+    * keeps only digital addresses listed in 'configured_address_types' parameter.
+    * groups response from /klantinteracties/api/v1/digitaleadressen endpoint by
+    the address type.
```

Comment:
```suggestion
    This function:

    * keeps only digital addresses listed in ``configured_address_types`` parameter.
    * groups response from ``/klantinteracties/api/v1/digitaleadressen`` endpoint by
      the address type.
```
Proper RST formatting :)